### PR TITLE
fix: config api

### PR DESCRIPTION
**Pull request by LaurenceLiZhixin**
Add configuration API of go sdk.

### CODE REVIEW

**Comment by daixiang0** [comment]
@LaurenceLiZhixin could add some tests?

File: client/configuration.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+	rsp, err := c.protoClient.GetConfiguration(ctx, &pb.GetConfigurationRequest{
+		StoreName: storeName,
+		Keys:      keys,
+		Metadata:  metadata,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	configItems := make([]*ConfigurationItem, 0)
```
Comment (daixiang0) [inline client/configuration.go:48]: Replace length of `rsp.Items` with `0`.

File: client/configuration.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	}
+	if len(items) == 0 {
+		return nil, nil
+	}
+	return items[0], nil
+}
+
+func (c *GRPCClient) GetConfigurationItems(ctx context.Context, storeName string, keys []string, opts ...ConfigurationOpt) ([]*ConfigurationItem, error) {
+	metadata := make(map[string]string)
+	for _, o := range opts {
```
Comment (daixiang0) [inline client/configuration.go:36]: Replace `opt` with `o`/

File: client/configuration.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+				return
+			}
+			configurationItems := make([]*ConfigurationItem, 0)
+			for _, v := range rsp.Items {
+				configurationItems = append(configurationItems, &ConfigurationItem{
+					Key:      v.Key,
+					Value:    v.Value,
+					Version:  v.Version,
+					Metadata: v.Metadata,
+				})
```
Comment (daixiang0) [inline client/configuration.go:98]: See those code block many times, covert them to a unit function.

**Comment by daixiang0** [comment]
@LaurenceLiZhixin ping

**Comment by LaurenceLiZhixin** [comment]
Got it. I'll keep working on it. @daixiang0

**Comment by LaurenceLiZhixin** [comment]
The test may passed after issue https://github.com/dapr/dapr/issues/3906 and related prs merged.

**Comment by Weidaicheng** [comment]
May I ask when is about to merge this PR, I really want to use configuration now, thx.

File: go.mod [hunk 4]
```diff
@@ -1,9 +1,9 @@
 module github.com/dapr/go-sdk
 
-go 1.16
+go 1.15
```
Comment (morefreeze) [inline go.mod:3]: Just curious is this downgrade mandatory?

Comment (LaurenceLiZhixin) [reply go.mod:3]: It's a typo I think, i would fix soon.

**Comment by LaurenceLiZhixin** [comment]
Just waiting for this pr https://github.com/dapr/dapr/pull/4130, and I would fix this pr as soon as possible.

File: client/client_test.go [hunk 5]
```diff
@@ -348,3 +356,62 @@ func (s *testDaprServer) UnregisterActorTimer(context.Context, *pb.UnregisterAct
 func (s *testDaprServer) Shutdown(ctx context.Context, req *empty.Empty) (*empty.Empty, error) {
 	return &empty.Empty{}, nil
 }
+
+func (s *testDaprServer) GetConfigurationAlpha1(ctx context.Context, in *pb.GetConfigurationRequest) (*pb.GetConfigurationResponse, error) {
+	items := make([]*commonv1pb.ConfigurationItem, 0)
```
Comment (daixiang0) [inline client/client_test.go:361]: Move it to L364.

File: actor/config/config.go [hunk 6]
```diff
@@ -35,8 +35,8 @@ func GetConfigFromOptions(opts ...Option) *ActorConfig {
 	conf := &ActorConfig{
 		SerializerType: constant.DefaultSerializerType,
 	}
-	for _, opt := range opts {
-		opt(conf)
+	for _, o := range opts {
```
[the remark this answers is not in the document]
Comment (LaurenceLiZhixin) [reply actor/config/config.go:38]: What about this comment? 
image

[the remark this answers is not in the document]
Comment (daixiang0) [reply actor/config/config.go:38]: Oops, seems I use wrong grammar...

File: client/client_test.go [hunk 7]
```diff
@@ -348,3 +356,62 @@ func (s *testDaprServer) UnregisterActorTimer(context.Context, *pb.UnregisterAct
...
+		}
+		items := make([]*commonv1pb.ConfigurationItem, 0)
+		for _, v := range in.GetKeys() {
+			items = append(items, &commonv1pb.ConfigurationItem{
+				Key:   v,
+				Value: v + "_" + strconv.Itoa(i),
+			},
+			)
+		}
+		if err := server.Send(&pb.SubscribeConfigurationResponse{
```
[the remark this answers is not in the document]
Comment (LaurenceLiZhixin) [reply client/client_test.go:396]: This is just for ut, I think just send once is enough to test subscribe feature now. @daixiang0

File: client/client_test.go (hunk 5 above)
Comment (LaurenceLiZhixin) [reply client/client_test.go:361]: done

File: actor/config/config.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (LaurenceLiZhixin) [reply actor/config/config.go:38]: I've changed to opt now. @daixiang0

**Comment by LaurenceLiZhixin** [comment]
ping @daixiang0

**Comment by LaurenceLiZhixin** [comment]
ping 
@daixiang0

**Comment by yaron2** [comment]
@daixiang0 can you review this? thanks!

**Review by daixiang0 — APPROVED** [review]
LGTM.

/cc @yaron2

**Comment by yaron2** [comment]
Thank you @LaurenceLiZhixin

**Pull request merged**